### PR TITLE
Wdfn 501 - Cleaned up the code, all tests run, tests were added for new functions

### PR DESCRIPTION
**Pull request by mbucknell**
Before making a pull request
----------------------------

- [X] Make sure all tests run
- [ ] Update the changelog appropriately

Title
-----------
Clean up refactor and got all tests except download-links in the hydrograph component running. Tests were added for new functions, both python and javascript

Description
-----------
Will annotate the PR.

After making a pull request
---------------------------
- [ ] If appropriate, put the link to the PR in the JIRA ticket
- [ ] Assign someone to review unless the change is trivial

### CODE REVIEW

File: assets/src/scripts/monitoring-location/components/hydrograph/data-table.test.js [hunk 1]
```diff
@@ -3,150 +3,12 @@ import {select} from 'd3-selection';
 import {configureStore} from 'ml/store';
 
 import {drawDataTables} from './data-table';
+import {TEST_PRIMARY_IV_DATA, TEST_GW_LEVELS} from './mock-hydrograph-state';
```
Comment (mbucknell) [inline assets/src/scripts/monitoring-location/components/hydrograph/data-table.test.js:6]: I put test data that could be reusued for javascript modules within components/hydrograph in mock-hydrograph-state. I think this should make it simpler to update tests as stored data changes.

File: assets/src/scripts/monitoring-location/components/hydrograph/index.test.js [hunk 2]
```diff
@@ -217,390 +83,220 @@ describe('monitoring-location/components/hydrograph module', () => {
...
-    });
-
-    describe('graphNode contains the expected elements when showOnlyGraph is false', () => {
-        /* eslint no-use-before-define: 0 */
-        let store;
-        let resolvedLoadPromise = Promise.resolve();
-        beforeEach(() => {
-            jest.spyOn(floodDataActions, 'retrieveWaterwatchData').mockReturnValue(function() {
-                return Promise.resolve({});
+            hydrographData.retrieveHydrographData = jest.fn(() => {
```
Comment (mbucknell) [inline assets/src/scripts/monitoring-location/components/hydrograph/index.test.js:271]: This is the function I was having trouble mocking. The asynchronous actions function return a function which returns a function that returns a promise. I just needed to really look at what the real asynchronous actions did.

**Review by mbucknell — COMMENTED** [review]
My testing philosophy for components/hydrograph was to really test thoroughly the selectors. For the rendering code, I tried to test any events that triggered a Redux action as well as testing initialization code and basic "did it render something".  My goal is to make the rendering code tests a little less brittle and yet really test the heck out of the selectors which are generally simpler.

File: assets/src/scripts/monitoring-location/components/hydrograph/data-table.test.js (hunk 1 above)
Comment (abriggs-usgs) [reply assets/src/scripts/monitoring-location/components/hydrograph/data-table.test.js:6]: Good idea. That will come in handy.

File: assets/src/scripts/monitoring-location/components/hydrograph/download-links.test.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ import {configureStore} from 'ml/store';
 import {renderDownloadLinks} from 'ivhydrograph/download-links';
 
 
-describe('monitoring-location/components/hydrograph/download-links', () => {
+xdescribe('monitoring-location/components/hydrograph/download-links', () => {
```
Comment (abriggs-usgs) [inline assets/src/scripts/monitoring-location/components/hydrograph/download-links.test.js:9]: Is this one intentionally skipped?

**Review by abriggs-usgs — APPROVED** [review]
Whew, that is a overwhelming amount of code! Hard to comprehend all the details, but I really like the focus on selectors and consolidating some of the mocked state into reusable units. Incredible work!!

File: assets/src/scripts/monitoring-location/components/hydrograph/download-links.test.js (hunk 3 above)
Comment (mbucknell) [reply assets/src/scripts/monitoring-location/components/hydrograph/download-links.test.js:9]: Yes. The download-links doesn't work. Fixing the tests is part of WDFN-500

**Pull request merged**